.github/workflows/composites/env-variables/action.yaml

```diff
@@ -6,5 +6,19 @@ runs:
     - name: set env variables
       shell: bash
       run: |
+        
+        baseRef=$GITHUB_BASE_REF
```

Comment:
this is by far not intuitive, but I hope that my comment here will make it clear why this is needed. We run the pipeline on two events : a PR and a push to the branch. In order to know if this needs to run on jdk-8 or jdk-17, I need to know what "base branch" is this going to. If it's `2.1.x` it must use `jdk-8`, if it's main, it must use jdk-17.

When this is a PR, things are easy, github provides: `GITHUB_BASE_REF`, so I know what branch is this PR supposed to go into. The problem is, that when this is a simple push (when you actually merge any PR), that variable is not available. So I am getting it from `${{ github.ref }}`. It needs a bit of parsing, cause it comes in the form of `xxx/yyy/<BRANCH>`.

I hope this makes sense.